src/eureka/S3_data_reduction/wfc3.py

```diff
@@ -747,27 +953,100 @@ def writeDrift2D(arg):
                                          ky=ky, s=0)
-        data.mask[n] = spline((iy-meta.drift2D[-1][n, 1] +
-                               meta.drift2D_int[-1][n, 1]).flatten(),
-                              (ix-meta.drift2D[-1][n, 0] +
-                               meta.drift2D_int[-1][n, 0]).flatten())
+        data.mask[n] = spline((iy-drift2D[n, 1] +
+                               drift2D_int[n, 1]).flatten(),
+                              (ix-drift2D[n, 0] +
+                               drift2D_int[n, 0]).flatten())
+        # Fractional masking won't work - make sure it is all integer
+        data.mask[n] = np.round(data.mask[n]).astype(int)
         spline = spi.RectBivariateSpline(iy, ix, data.variance[n], kx=kx,
                                          ky=ky, s=0)
-        data.variance[n] = spline((iy-meta.drift2D[-1][n, 1] +
-                                   meta.drift2D_int[-1][n, 1]).flatten(),
-                                  (ix-meta.drift2D[-1][n, 0] +
-                                   meta.drift2D_int[-1][n, 0]).flatten())
+        data.variance[n] = spline((iy-drift2D[n, 1] +
+                                   drift2D_int[n, 1]).flatten(),
+                                  (ix-drift2D[n, 0] +
+                                   drift2D_int[n, 0]).flatten())
         spline = spi.RectBivariateSpline(iy, ix, data.bg[n], kx=kx,
                                          ky=ky, s=0)
-        data.bg[n] = spline((iy-meta.drift2D[-1][n, 1] +
-                             meta.drift2D_int[-1][n, 1]).flatten(),
-                            (ix-meta.drift2D[-1][n, 0] +
-                             meta.drift2D_int[-1][n, 0]).flatten())
+        data.bg[n] = spline((iy-drift2D[n, 1] +
+                             drift2D_int[n, 1]).flatten(),
+                            (ix-drift2D[n, 0] +
+                             drift2D_int[n, 0]).flatten())
 
-    return data, meta
+    return data, meta, log
+
+
+def cut_aperture(data, meta, log):
+    """Select the aperture region out of each trimmed image.
+
+    Parameters
+    ----------
+    data : Xarray Dataset
+        The Dataset object.
+    meta : eureka.lib.readECF.MetaClass
+        The metadata object.
+    log : logedit.Logedit
+        The current log.
+
+    Returns
+    -------
+    apdata : ndarray
+        The flux values over the aperture region.
+    aperr : ndarray
+        The noise values over the aperture region.
+    apmask : ndarray
+        The mask values over the aperture region.
+    apbg : ndarray
+        The background flux values over the aperture region.
+    apv0 : ndarray
+        The v0 values over the aperture region.
+
+    Notes
+    -----
+    History:
+
+    - 2022-06-17, Taylor J Bell
+        Initial version, edited to work for HST scanned observations.
+    """
+    log.writelog('  Extracting aperture region',
+                 mute=(not meta.verbose))
+
+    apdata = np.zeros((data.flux.shape[0], meta.spec_hw*2,
```

Comment:
Unless best practices have changed recently, we should be calling `data.flux.shape` only once and not 10 times.